src/lib.rs

```diff
@@ -164,6 +167,41 @@ pub trait Uncertain {
         BoxedUncertain::new(self)
     }
 
+    /// Bundle this uncertain value with a cache, so it can be reused in a calculation.
```

Comment:
Not sure about the wording here. What do you think of:

```
Wrap this value with a cache, so that it can be reused within a calculation.

Normally, uncertain values do not implement `Copy` or `Clone`, since the same value is only allowed to be samples once for every epoch. The cache added by this wrapper allows a value to be reused, by caching the sample result for every epoch and implementing [`Uncertain`] for references as well.

... examples etc ... 
```